bin/build.js

```diff
@@ -1,43 +1,108 @@
-const { build } = require('esbuild')
+const { build, context } = require('esbuild')
 const { dependencies, peerDependencies } = require('../package.json')
 const { Generator } = require('npm-dts')
 const { sassPlugin } = require('esbuild-sass-plugin')
+const { rename, rm } = require('node:fs/promises')
+
+const OUT_DIR = 'dist'
+
+const args = process.argv.slice(2)
+const watchModeEnabled = args.includes('--watch')
 
 new Generator({
   entry: 'index.tsx',
-  output: 'dist/index.d.ts',
+  output: `${OUT_DIR}/index.d.ts`,
   tsc: '-p ./tsconfig.json',
-}).generate()
-
-const entryPoints = ['src/index.tsx']
+},
+  null,
+  !watchModeEnabled
+)
+.generate()
+.catch(() => {
+  throw new Error(
+    [
+      "Failed to generate types:",
+      "  - Try running `npx npm-dts generate -L debug` for more information",
+    ].join('\n'),
+  )
+})
 
-/**
- * @type {import('esbuild').BuildOptions}
- */
+/** @type {import('esbuild').BuildOptions} */
 const sharedConfig = {
-  entryPoints,
+  entryPoints: [ 'src/index.tsx' ],
   bundle: true,
   minify: false,
-  sourcemap: true,
   tsconfig: './tsconfig.json',
   target: 'es2016',
   external: [
-    ...Object.keys(dependencies || {}),
     ...Object.keys(peerDependencies || {}),
+    ...Object.keys(dependencies || {}),
   ],
-  plugins: [sassPlugin()],
+  plugins: [ sassPlugin() ],
 }
 
-build({
+/** @type {import('esbuild').BuildOptions} */
+const cjsConfig = {
   ...sharedConfig,
-  entryPoints: [...entryPoints, 'src/styles/index.scss'],
-  platform: 'node', // for CJS
-  outdir: 'dist',
-})
+  platform: 'node',
+  outdir: `${OUT_DIR}/cjs`,
+  outExtension: { '.js': '.cjs' },
+}
 
-build({
+const STYLE_ENTRY_POINT = 'src/styles/index.scss'
+
+function moveStyles(baseDir, { withMap = false } = {}) {
+  return Promise.all(
+    [
+      rename(`${baseDir}/styles/index.css`, `${OUT_DIR}/index.css`),
+      withMap
+        ? rename(`${baseDir}/styles/index.css.map`, `${OUT_DIR}/index.css.map`)
+        : null
+    ].filter(Boolean)
+  )
+  .then(() => rm(`${baseDir}/styles`, { recursive: true }))
+}
```

Comment:
Am I understanding correctly that the way we can make it easier to import the CSS is to add to build step to literally just move the file?

Reply:
Similar to our exported types, our CSS is not specific to ESM or CJS.  It happens to get initially nested inside a certain sub-folder, so I explicitly move it out of there.

The change in `package.json` (see `"exports" and "style"`) should make it so our css files end-up in a standard location.